docs/source/basic/terms.rst

```diff
@@ -0,0 +1,57 @@
+Terms
+=====
+
+This page provides an overview of the terms used in ``alpaka`` and the relationships between them.
+
+.. image:: /images/alpaka_terms.svg
+
+Platform
+--------
+
+- A ``platform`` contains information about the system, e.g. the available devices. 
+- Depending on the platform, it also contains a runtime context.
+- A ``platform`` has a least one device but it can also has many device.
+- Each ``platform`` can be used with N ``accelerator``. ``platforms`` and ``accelerator`` cannot freely combined. An ``accelerator`` supports only a specific ``platform``.
+ 
+Device
+------
+
+- A ``device`` represent a compute unit, such as a CPU or a GPU.
+- Each ``device`` is bounded to a specific ``platform``.
+- Each ``device`` can have N ``queues``.
+
+Accelerator
+-----------
+
+- A ``accelerator`` is a index mapping function. It distributes the index space to the chunks. The ``accelerator`` maps a continues index space to a blocked index domain decomposition.
+- It is not allowed to create an instance of an ``accelerator``.
+- A ``accelerator`` is bounded to a specific ``platform``.
+
+Queue
+-----
+
```

Comment:
I would add that a queue can be either "synchronous" (blocking) or "asynchronous" (non-blocking), and that this may determine how the work is submitted to the device, and if a synchronisation is performed after each operation.